pkg/evaluator/builtin.go

```diff
@@ -597,6 +595,71 @@ func dashFunc(dashFn func([]float64)) BuiltinFunc {
 	}
 }
 
+var fontDecl = &parser.FuncDeclStmt{
+	Name:       "font",
+	Params:     []*parser.Var{{Name: "properties", T: parser.GENERIC_MAP}},
+	ReturnType: parser.NONE_TYPE,
+}
+
+func parseFontProps(arg *Map) (map[string]any, error) {
+	props := map[string]any{}
+	propTypes := map[string]string{
+		"family":        "string",
+		"size":          "num",
+		"weight":        "num",
+		"style":         "string",
+		"baseline":      "string",
+		"align":         "string",
+		"letterspacing": "num",
+	}
+	for key, val := range arg.Pairs {
+		propType, ok := propTypes[key]
+		if !ok {
+			return nil, fmt.Errorf("%w: unknown property %q", ErrBadArguments, key)
+		}
+		anyVal, ok := val.(*Any)
+		if ok {
```

Comment:
nit: I'd combine these lines because `anyVal` and `ok` are not needed after this:

    if a, ok := val.(*Any); ok {
        val = a.Val
    }

Reply:
ok, agreed.

Reply:
done.